src/lib/comms/sol-lwm2m-client.c

```diff
@@ -1567,12 +1594,28 @@ sol_lwm2m_client_new(const char *name, const char *path, const char *sms,
     client->coap_server = sol_coap_server_new(&servaddr, false);
     SOL_NULL_CHECK_GOTO(client->coap_server, err_coap);
 
+    client->dtls_server = sol_coap_server_new(&servaddr, true);
+    if (!client->dtls_server) {
+        if (errno == ENOSYS) {
```

Comment:
No... If DTLS is not supported, that function will return with `errno` set to ENOSYS, in which case we just print a message using `SOL_INF`. Otherwise, DTLS is indeed supported and the error was due to something else; in that case, we use `SOL_WRN` and `goto err_dtls`.

This is similar to https://github.com/bsmelo/soletta/blob/lwm2m-dtls_psk-v1/src/lib/comms/sol-oic-server.c#L615.